### DISCUSSION TITLE
Graphviz dot variable substition

### DISCUSSION THREAD

**Discussion by dmenne** [quarto-dev/quarto-cli#8181] [Q&A]
### Description

Using DiagrammR, I can do variable substitution with `glue` or the `@@1` syntax. I have not found a valid syntax to do this with an explicit `dot` chunk.

````
  ```{r}
  library(DiagrammeR)
  library(glue)
  n_recruited = 50
  digraph = glue(
    'digraph dg {{
      graph[layout = dot] 
      A[label = "n={n_recruited}"]  
    }}' )
  
   grViz(digraph )
  ```
  
  ```{r}
  digraph = glue(
    "digraph dg {{
      graph[layout = dot] 
      A[label = @@1]  
    }}
    [1]: n_recruited  " )
  
   grViz(digraph )
  ```
  
  Chunk below is just a placeholder for  a non-working syntax
  
  ```{dot}
  digraph dg {
    graph[layout = dot] 
    A[label = n_recruited]  
  } 
  ```

````

**Comment by mcanouil** [comment]
Graphviz does not support variable substitution.
It is possible in `DiagrammeR` only because of R.

Note: this is not related to Quarto/Pandoc/Rmarkdown.

**Reply by dmenne** [reply]
I agree that is not related to Q/P/R, but maybe it would be good if this were documented; I spent quite a bit of time to be reasonably sure the failure is not due to my ignorance.

**Reply by mcanouil** [reply]
Quarto tries to document when the implementation of those third party tools inside it, means additional limitations to the third party tool ones.
Here variable substitution is a 100 % GraphViz limitation (and there are plenty more especially regarding the type of diagram you can make).
I don't think Quarto should document all third parties limitations as it would mean to do so for R packages, Python modules, Julia libraries, etc. I'm afraid, this is not at all doable.
GraphViz has its own documentation which you can read, see <https://graphviz.org/>.

Note: To answer you, I searched through GraphViz documentation.

**Comment by cderv** [comment]
For reference, here is the doc for this **DiagrammeR** specific feature: 
https://rich-iannone.github.io/DiagrammeR/articles/graphviz-mermaid.html#graphviz-substitution

Using R and this package seems the way to go to have this feature available within Quarto.

**Comment by harrelfe** [comment]
Variable substitution, including whole data frames, is done with `graphviz` using the R `qreport` `makegraphviz` function illustrated [here](https://hbiostat.org/rflow/rformat#sec-rformat-diagrams).
